### PR TITLE
fix(sandbox): diagnose legacy mounts under declared agent config

### LINKED ISSUE #3775: agent_config_dir sandbox layout changed with no diagnostic: existing setups open on the folder-trust dialog
#3678 (`b2569c75e`) moved the sandbox store for a declared `session.agent_config_dir` from a fixed `<dir>/sandbox` to a per-session `<dir>/sandbox-v2/<instance-id>`, and AoE now mounts that store at the built-in config path and exports the config-dir variable itself. The reference table in `docs/guides/configuration.md` records the new layout. Nothing tells an existing user their setup changed.

Under the old layout the user supplied the mount, because AoE did not, and a wrapper agent exporting `CLAUDE_CONFIG_DIR` pointed at it. After the change that wrapper overrides the variable AoE just set, discarding the store AoE seeded, including its folder-trust record. The session opens on the folder-trust dialog with nothing logged. The status change reads `rule=folder_trust_prompt` and no warning is emitted.

The same commit removed `warn_unless_mounted` and `is_mounted_into_container`, which was the only diagnostic covering a declared directory that reaches no container.

Per AGENTS.md a breaking change needs no shim, so the ask is a diagnostic and an upgrade note, not compatibility. Warning once when a declared `agent_config_dir` still has a `sandbox.extra_volumes` entry mounting under it would name the cause at the point it bites.

_Note: this issue was drafted by Claude Opus 5 via back-and-forth with @njbrake. The reasoning and decisions are his; the prose is Claude's._

### PR DESCRIPTION
**Pull request by jerome-benoit**
## Description

Fixes #3775.

Add a diagnostic and an upgrade note for manual sandbox mounts retained after #3678. No compatibility behavior: mounts, environment variables, trust seeding, and persisted stores are unchanged.

### Root cause and evidence

#3678 moved a declared agent config store from `<dir>/sandbox` to `<dir>/sandbox-v2/<instance-id>`, mounted at the built-in config path. A wrapper's later `CLAUDE_CONFIG_DIR` export can select the old, separately mounted store instead of the one AoE prepared.

Reproduced locally on base `8876585e`, AoE 1.15.3, Linux, Docker 29.8.0, and sandbox-image Claude Code 2.1.258. The fixture used an isolated HOME, synthetic onboarding state, no real credentials, a declared account directory, and `<account>/sandbox:/root/legacy-account`. The wrapper exported `CLAUDE_CONFIG_DIR=/root/legacy-account`.

Observed:

```text
Yes, I trust this folder
WARN count: 0
managed trust: True
wrapper trust: None
```

Changing only the wrapper export to `/root/.claude` removed the folder-trust dialog. Restoring `/root/legacy-account` brought it back, without accepting trust or removing the manual mount. This isolates the selected config path, rather than missing trust seeding or a mount at the same destination.

The old `warn_unless_mounted` checked for a missing mount. It would have accepted this old recipe before the layout change. This PR adds a diagnostic for a retained manual recipe; it does not restore that old predicate.

Relevant source: `src/session/instance/container.rs:141,321-348`; existing store/seed behavior in `src/session/config/container_config.rs:1982-2013,2232-2267`.

### Fix

- Warn once per `get_container_for_instance` preparation when the first parsable extra-volume source equals or lies under the selected agent's declared config directory.
- Name the agent, declared directory, and manual volume. Describe a possible bypass of per-session isolation and staged folder trust, not a proven wrapper failure.
- Keep emission outside the config builder: Pi may build the config again while preparing its launch.
- Add the upgrade note to the existing **Per-session agent stores** section. Preserve AoE's config-dir variables inside the sandbox; retain host-only account selection outside it.

### Verification

The same full-binary warning assertion was run before and after the fix:

```text
Before: named warning count: 0
AssertionError: expected one named diagnostic, got 0
After:  named warning count: 1
```

The emitted WARN names `sandbox.extra_volumes`, `agent_config_dir`, the fixture agent, and the legacy mount. The unchanged wrapper still displays folder trust: this is diagnostic-only.

- `cargo test --lib legacy_agent_config_mount_warns_once_for_selected_profile_and_agent`: 1 passed.
- `cargo test --lib session::instance::container::tests`: 2 passed.
- `cargo test --lib session::config::container_config::tests`: 122 passed.
- `cargo test`: 6939 passed, 8 ignored.
- `cargo build`, `cargo clippy`, and `cargo fmt`: passed.
- Commit hooks also passed `cargo clippy -- -D warnings` and `cargo fmt -- --check`.
- Live Pi 0.84.4 sandbox launch with two suspect volumes: exactly one named warning. Its native session ID was not reported within the launch observation window; this check does not claim native resume works.

The permanent Rust regression covers multiple matching mounts, the root itself, component boundaries, source versus destination, malformed entries, the selected profile and agent, repeated config builds, and leaving the legacy JSON untouched. The red/green integration assertion used the actual binary; the newly added private helper is not claimed to compile on the pre-fix revision.

### Risks and limits

A read-only reference mount may be legitimate, so this is a warning, never a rejection. Path matching is lexical and component-based; it does not resolve symlink aliases or normalize `..`. The diagnostic examines declared configuration, not the wrapper's actual file reads or the container's effective mounts. Each explicit container preparation can warn again; there is no process-global or persisted suppression state, and no polling-loop emission. One additional profile configuration resolution is performed per preparation.

## PR Type

- [ ] New Feature
- [x] Bug Fix
- [ ] Refactor
- [x] Documentation
- [ ] Infrastructure / CI

## Checklist

- [x] New and existing tests pass
- [x] Documentation was updated where necessary
- [ ] For UI changes: included screenshot or recording (N/A: logging diagnostic only; no rendering change)

## Test Coverage Analysis

**Web dashboard / structured view (Playwright user story)**
- [x] N/A: this PR doesn't change a user-facing dashboard flow
- [ ] Added or updated a Playwright spec under `web/tests/` and updated `web/tests/coverage-matrix.json`
- [ ] Skipped a test, because:

**TUI / CLI (e2e test)**
- [ ] N/A: this PR doesn't change TUI rendering, a CLI subcommand, or session lifecycle
- [ ] Added or updated an e2e test under `tests/e2e/`
- [x] Skipped a test, because: the warning contract is covered by an isolated in-module Rust regression and actual Claude/Pi sandbox smoke checks. A permanent Docker/agent-dependent test would be more expensive and less deterministic.

## AI Usage

- [ ] No AI was used
- [x] AI was used

**AI Model/Tool used:**
Oh My Pi, `openai-codex/gpt-6-astra`, with analysis-only subagents for independent RCA and design review.

**Any Additional AI Details you'd like to share:**
One agent applied all code, test, and documentation changes. Reproduction and verification commands were executed locally. This PR is intentionally a draft; it has not been merged.

**NOTE:**
When responding to reviewer questions, please respond yourself rather than copy/pasting reviewer comments into an AI and pasting back its answer. We want to discuss with you, not your AI :)

- [x] I am an AI Agent filling out this form (check box if true)

## Summary

- Added a one-time warning for manual sandbox mounts under a declared agent configuration directory.
- Added regression tests for matching, boundary, unrelated-agent, and duplicate-warning cases.
- Updated wrapper upgrade guidance to preserve AoE-managed config-directory variables and avoid outdated stores.

## Benefits

- Users can detect mounts that bypass per-session isolation or folder-trust state.
- Clear upgrade guidance helps prevent unexpected folder-trust prompts.

## Potential follow-up

- Monitor support reports for other upgrade patterns that may need documentation.

### CODE REVIEW

File: src/session/instance/container.rs [hunk 1]
```diff
@@ -317,6 +318,35 @@ impl Instance {
...
+        else {
+            return;
+        };
+        let Some(directory) = config.session.agent_config_dir_for(&self.tool, &home) else {
+            return;
+        };
+        if let Some(entry) = config.sandbox.extra_volumes.iter().find(|entry| {
+            entry
+                .split_once(':')
+                .is_some_and(|(source, _)| Path::new(source).starts_with(&directory))
```
[the remark this answers is not in the document]
Comment (jerome-benoit) [reply src/session/instance/container.rs:338]: **Out of scope: native Windows is not supported.**

- `README.md:127–129` explicitly limits Windows support to WSL2, where AoE uses Linux paths such as `/mnt/c/Users/alice/.claude`.
- The existing volume builder also splits at the first colon (`src/session/config/container_config.rs:2249–2262`). A native drive-letter source is not supported by that path either.
- The new diagnostic handles the corresponding Linux/WSL path. Adding drive-letter parsing only here would diverge from the actual mount parser without enabling native Windows support.

No code change is warranted for this finding.

**Review by njbrake — APPROVED** [review]
Approve.

The diagnostic mirrors `validate_managed_container_environment`, the sibling guardrail for `sandbox.environment`, and warns rather than bails because a read-only reference mount under the config dir can be legitimate. The call site is the chokepoint every sandboxed launch passes through.

Verified the regression test is load-bearing: swapping `Path::starts_with` for a naive string prefix makes the `account-other` boundary case fail.

Agree on the Windows drive-letter dismissal. `container_config.rs:2249` splits at the first colon too, so parsing drive letters only here would diverge from the parser this mirrors.

Pushed two commits: the upgrade note restated guidance already at `sandbox.md:199-204` and `configuration.md:227-231`, so it is folded into the first; plus a comment recording why the check sits outside the config builder.

_Note: this review was drafted by Claude Opus 5 via back-and-forth with @njbrake. The reasoning and decisions are his; the prose is Claude's._
